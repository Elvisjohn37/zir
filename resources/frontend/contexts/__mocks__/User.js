import React from 'react';
import { componentModuleMock, transformProps } from 'frontend/tests/helpers';
let mockedModules = componentModuleMock(jest.createMockFromModule('../User'));

mockedModules.default = ({ children, ...props }) => {
	let { newProps, childProps } = transformProps(props);

	return (
		<div {...newProps} data-testid="User">
			{children}
			{childProps.map(() => childProps)}
		</div>
	);
};

let userState;
mockedModules.userDispatch = () => {};

// main component
mockedModules.UserConsumer = ({ children }) => {
	return children({ userState, userDispatch: mockedModules.userDispatch });
};

// setter
mockedModules.setUserState = (state) => {
	Object.assign(userState, state);
};

// module resetter
mockedModules.resetUserState = () => {
	userState = {
		isLoading: true,
		isLogin: false,
		isError: false,
		user: {
			parent: 'testParent',
		},
	};
};
mockedModules.resetUserState();

// getter
mockedModules.getUserState = () => {
	return userState;
};

module.exports = mockedModules;
