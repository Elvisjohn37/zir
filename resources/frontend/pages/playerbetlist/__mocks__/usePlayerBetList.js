import React from 'react';
import {
	getPlayerPromoList,
	getPlayerBetList,
	setPlayerBetListData,
	resetPlayerBetListData,
	getPlayerBetListData,
} from 'frontend/ajax/__mocks__/backend';

let mockedModule = jest.createMockFromModule('../usePlayerBetList.js');

let usePlayerBetListData;
let isProceedHook = true;

mockedModule.usePlayerBetList = jest.fn(
	({ isFetching, setIsFetching, date, page, setIsPromoBettingListError, type, productID }) => {
		let [playerPromoBetList, setPlayerPromoBetList] = React.useState({
			content: [],
			totalRow: [],
			rowsPerPage: 0,
			product: '',
		});

		if (!isProceedHook) {
			return playerPromoBetList;
		}

		React.useEffect(() => {
			if (isFetching) {
				if (type == 'betting') {
					getPlayerBetList({
						date,
						productID,
						page,
						success: (response) => {
							setIsFetching(false);
							usePlayerBetListData = response.data.data;
							setPlayerPromoBetList(response.data.data);
						},
						error: () => {
							setIsPromoBettingListError(true);
							setIsFetching(false);
						},
					});
				} else {
					getPlayerPromoList({
						date,
						page,
						success: (response) => {
							setIsFetching(false);
							usePlayerBetListData = response.data.data;
							setPlayerPromoBetList(response.data.data);
						},
						error: () => {
							setIsPromoBettingListError(true);
							setIsFetching(false);
						},
					});
				}
			}
		}, [isFetching]);
		return playerPromoBetList;
	}
);

function setIsProceedHook(isProceed) {
	isProceedHook = isProceed;
}

function setUsePlayerBetListData(data) {
	usePlayerBetListData = { ...data };
}

function resetUsePlayerBetListData() {
	usePlayerBetListData = {};
	resetPlayerBetListData();
	isProceedHook = true;
}

function setUsePlayerBetListResponse(result, data) {
	setPlayerBetListData(result, data);
}

export default mockedModule.usePlayerBetList;

export {
	setUsePlayerBetListData,
	resetUsePlayerBetListData,
	usePlayerBetListData,
	setUsePlayerBetListResponse,
	setIsProceedHook,
	getPlayerBetListData,
};
