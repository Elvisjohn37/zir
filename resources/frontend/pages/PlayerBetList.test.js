import React from 'react';
import { render, screen, cleanup, within } from '@testing-library/react';
import PlayerBetList from './PlayerBetList';
import { setUserState } from 'frontend/contexts/User';
import * as usePlayerBetList from './playerbetlist/usePlayerBetList';
import * as MediaQuery from 'react-responsive';
import userEvent from '@testing-library/user-event';

jest.mock('@mui/material/Table');
jest.mock('@mui/material/TableHead');
jest.mock('@mui/material/TableRow');
jest.mock('@mui/material/TableBody');
jest.mock('@mui/material/TableCell');
jest.mock('@mui/material/TableContainer');
jest.mock('frontend/components/TableIndicator');
jest.mock('frontend/components/TableRefresh');
jest.mock('frontend/components/Sidebar');
jest.mock('frontend/components/Language');
jest.mock('frontend/components/ReportCommonFormatter');
jest.mock('frontend/components/ReportBetFormatter');
jest.mock('frontend/contexts/Layout');
jest.mock('frontend/contexts/User');
jest.mock('./playerbetlist/usePlayerBetList');
jest.mock('react-responsive');

describe('<PlayerBetList />', () => {
	afterEach(() => {
		resetMocks();
		cleanup();
	});

	function resetMocks() {
		jest.clearAllMocks();
		jest.restoreAllMocks();
		usePlayerBetList.resetUsePlayerBetListData();
		MediaQuery.setMediaQuery(true);
	}

	function getComponentProps(transactionType = 'betting') {
		let params = {
			type: transactionType,
			date: '2021-07-15',
		};
		if (transactionType == 'betting') {
			params = { ...params, productID: 1 };
		}
		return params;
	}

	function expectCorrectTableContent() {
		setUserState({ user: { parent: 'test' } });
		let responseData = usePlayerBetList.getPlayerBetListData().data.data;
		let columnsLength = 7;
		responseData.content.forEach((data, index) => {
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 1]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 2]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 3]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 4]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 5]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 6]).toBeInTheDocument();
		});
	}

	it('should match snapshot', () => {
		let { asFragment } = render(<PlayerBetList match={{ params: getComponentProps() }} />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should have table loading indicator on table during data fetching', () => {
		usePlayerBetList.setIsProceedHook(false);
		render(<PlayerBetList match={{ params: getComponentProps() }} />);
		expect(screen.getByText('Loading TableIndicator')).toBeInTheDocument();
	});

	it('should fetch correct data when player select promotion or bet list report type', () => {
		let { rerender } = render(<PlayerBetList match={{ params: getComponentProps() }} />);
		rerender(<PlayerBetList match={{ params: getComponentProps('promotion') }} />);
		expect(usePlayerBetList.default).toHaveBeenNthCalledWith(1, {
			isFetching: true,
			setIsFetching: expect.any(Function),
			date: '2021-07-15',
			page: 1,
			productID: 1,
			setIsPromoBettingListError: expect.any(Function),
			type: 'betting',
		});
		expect(usePlayerBetList.default).toHaveBeenNthCalledWith(2, {
			isFetching: false,
			setIsFetching: expect.any(Function),
			date: '2021-07-15',
			page: 1,
			setIsPromoBettingListError: expect.any(Function),
			type: 'betting',
			productID: 1,
		});
		expect(usePlayerBetList.default).toHaveBeenNthCalledWith(3, {
			isFetching: false,
			setIsFetching: expect.any(Function),
			date: '2021-07-15',
			page: 1,
			setIsPromoBettingListError: expect.any(Function),
			type: 'promotion',
			productID: undefined,
		});
	});

	it('should display correct data in table after request from backend', () => {
		render(<PlayerBetList match={{ params: getComponentProps() }} />);
		expectCorrectTableContent();
		usePlayerBetList.setUsePlayerBetListResponse(true, {
			content: [
				{
					rowNo: 100,
					betLinkFormat: 'hasBetLink',
					commission: '0.000000',
					dateTime: '2021-09-08 11:19:51.000000',
					event: 'W',
					gameID: '00014',
					gameName: 'Djap Go',
					grossRake: '0.000000',
					memWLCommisionFormatter: 'withMemWLCommission',
					netwin: '10000.000000',
					reasonFormat: 'noReason',
					roundDetID: 2,
					roundIDFormat: 'roundDetID',
					stake: '4.000000',
					tableNameFormat: 'gameName',
					transactionDetID: 'zW9',
					turnover: '4.000000',
				},
				{
					rowNo: 101,
					betLinkFormat: 'hasBetLink',
					commission: '0.000000',
					dateTime: '2021-09-08 11:19:51.000000',
					event: 'L',
					gameID: '00014',
					gameName: 'Djap Go',
					grossRake: '0.000000',
					memWLCommisionFormatter: 'withMemWLCommission',
					netwin: '10000.000000',
					reasonFormat: 'noReason',
					roundDetID: 2,
					roundIDFormat: 'roundDetID',
					stake: '4.000000',
					tableNameFormat: 'gameName',
					transactionDetID: 'zW9',
					turnover: '4.000000',
				},
			],
		});
		cleanup();
		resetMocks();
		render(<PlayerBetList match={{ params: getComponentProps() }} />);
		expectCorrectTableContent();
	});

	it('should fetch data from backend when user click the refresh button', () => {
		render(<PlayerBetList match={{ params: getComponentProps() }} />);
		userEvent.click(screen.getByTestId('MobileTableRefresh'));
		expect(usePlayerBetList.default).toHaveBeenNthCalledWith(1, {
			isFetching: true,
			setIsFetching: expect.any(Function),
			date: '2021-07-15',
			page: 1,
			productID: 1,
			setIsPromoBettingListError: expect.any(Function),
			type: 'betting',
		});
		expect(usePlayerBetList.default).toHaveBeenNthCalledWith(2, {
			isFetching: false,
			setIsFetching: expect.any(Function),
			date: '2021-07-15',
			page: 1,
			productID: 1,
			setIsPromoBettingListError: expect.any(Function),
			type: 'betting',
		});
		expect(usePlayerBetList.default).toHaveBeenNthCalledWith(3, {
			isFetching: true,
			setIsFetching: expect.any(Function),
			date: '2021-07-15',
			page: 1,
			productID: 1,
			setIsPromoBettingListError: expect.any(Function),
			type: 'betting',
		});
		expect(usePlayerBetList.default).toHaveBeenNthCalledWith(4, {
			isFetching: false,
			setIsFetching: expect.any(Function),
			date: '2021-07-15',
			page: 1,
			productID: 1,
			setIsPromoBettingListError: expect.any(Function),
			type: 'betting',
		});
		expect(usePlayerBetList.default).toHaveBeenCalledTimes(4);
	});

	it('should display refresh button base on the type of device', () => {
		let { rerender } = render(<PlayerBetList match={{ params: getComponentProps() }} />);
		let MobileTableRefresh = screen.getByTestId('MobileTableRefresh');
		expect(MobileTableRefresh).toBeInTheDocument();

		MediaQuery.setMediaQuery(false);
		rerender(<PlayerBetList match={{ params: getComponentProps() }} />);
		let DesktopTableRefresh = screen.getByTestId('DesktopTableRefresh');
		expect(DesktopTableRefresh).toBeInTheDocument();
	});

	it('should display pagination and record base on selected page', () => {
		render(<PlayerBetList match={{ params: getComponentProps() }} />);
		expect(screen.getByText('Pagination page: 0')).toBeInTheDocument();
		expect(
			screen.getByText('Pagination count: ' + usePlayerBetList.usePlayerBetListData.totalRow)
		).toBeInTheDocument();
		expect(
			screen.getByText('Pagination rowsPerPage: ' + usePlayerBetList.usePlayerBetListData.rowsPerPage)
		).toBeInTheDocument();
	});

	it('should display no record found when record is empty', () => {
		usePlayerBetList.setUsePlayerBetListResponse(true, { content: [] });
		let { container } = render(<PlayerBetList match={{ params: getComponentProps() }} />);
		let tableRow = container.getElementsByClassName('testTableRow');
		expect(tableRow).toHaveLength(0);
	});

	it('should display error message when backend response is not valid', () => {
		usePlayerBetList.setUsePlayerBetListResponse(false);
		render(<PlayerBetList match={{ params: getComponentProps() }} />);
		expect(screen.getByText('Error Response')).toBeInTheDocument();
	});
});
