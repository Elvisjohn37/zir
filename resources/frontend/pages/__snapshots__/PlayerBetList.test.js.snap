// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerBetList /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    category="reports"
    data-testid="LangDataProvider"
  >
    <div>
      reports
    </div>
    <div
      data-testid="default"
      sidebargroup="report"
    >
      <div
        class="tableSection"
      >
        <div
          data-testid="MobileTableRefresh"
          isdisabled="false"
        />
        <div>
          <div
            data-testid="TableIndicator"
            id="testTableInit"
            isempty="false"
            iserror="false"
            isloading="false"
            pagination="{\\"count\\":3,\\"page\\":0,\\"rowsPerPage\\":20}"
          >
            <span>
              Pagination count: 3
            </span>
            <span>
              Pagination page: 0
            </span>
            <span>
              Pagination rowsPerPage: 20
            </span>
            <span>
              Done loading TableIndicator
            </span>
            <div
              data-testid="TableContainer"
            >
              <div
                aria-label="a dense table"
                classes="{\\"root\\":\\"table\\"}"
                data-testid="Table"
              >
                <div
                  data-testid="TableHead"
                >
                  <div
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="colHeaderRowNum"
                      data-testid="TableCell"
                    >
                      #
                    </div>
                    <div
                      align="center"
                      class="colHeaderDateTime"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="date"
                      >
                        <div>
                          date
                        </div>
                      </div>
                       & 
                      <div
                        data-testid="FormattedMessage"
                        id="time"
                      >
                        <div>
                          time
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colHeaderGameDetails"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="gameDetails"
                      >
                        <div>
                          gameDetails
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colHeaderGameResult"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="gameResult"
                      >
                        <div>
                          gameResult
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colHeaderStakeTurnover"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="stakeTurnover"
                      >
                        <div>
                          stakeTurnover
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colHeaderGrossRake"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="grossRake"
                      >
                        <div>
                          grossRake
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colHeaderMemWLCommission"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="memWLCommission"
                      >
                        <div>
                          memWLCommission
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  data-testid="TableBody"
                >
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      1
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testGameDetails"
                        data-testid="GameDetails"
                        row="{\\"rowNo\\":1,\\"betLinkFormat\\":\\"hasBetLink0\\",\\"commission\\":\\"0.0000000\\",\\"dateTime\\":\\"2021-09-08 11:19:51.0000000\\",\\"event\\":\\"W0\\",\\"gameID\\":\\"000140\\",\\"gameName\\":\\"Djap Go0\\",\\"grossRake\\":\\"0.0000000\\",\\"memWLCommisionFormatter\\":\\"withMemWLCommission0\\",\\"netwin\\":\\"10000.0000000\\",\\"reasonFormat\\":\\"noReason0\\",\\"roundDetID\\":2,\\"roundIDFormat\\":\\"roundDetID0\\",\\"stake\\":\\"4.0000000\\",\\"tableNameFormat\\":\\"gameName0\\",\\"transactionDetID\\":\\"zW90\\",\\"turnover\\":\\"4.0000000\\"}"
                      >
                        <span>
                          {"rowNo":1,"betLinkFormat":"hasBetLink0","commission":"0.0000000","dateTime":"2021-09-08 11:19:51.0000000","event":"W0","gameID":"000140","gameName":"Djap Go0","grossRake":"0.0000000","memWLCommisionFormatter":"withMemWLCommission0","netwin":"10000.0000000","reasonFormat":"noReason0","roundDetID":2,"roundIDFormat":"roundDetID0","stake":"4.0000000","tableNameFormat":"gameName0","transactionDetID":"zW90","turnover":"4.0000000"}
                        </span>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testGameResult"
                        data-testid="GameResult"
                        row="{\\"rowNo\\":1,\\"betLinkFormat\\":\\"hasBetLink0\\",\\"commission\\":\\"0.0000000\\",\\"dateTime\\":\\"2021-09-08 11:19:51.0000000\\",\\"event\\":\\"W0\\",\\"gameID\\":\\"000140\\",\\"gameName\\":\\"Djap Go0\\",\\"grossRake\\":\\"0.0000000\\",\\"memWLCommisionFormatter\\":\\"withMemWLCommission0\\",\\"netwin\\":\\"10000.0000000\\",\\"reasonFormat\\":\\"noReason0\\",\\"roundDetID\\":2,\\"roundIDFormat\\":\\"roundDetID0\\",\\"stake\\":\\"4.0000000\\",\\"tableNameFormat\\":\\"gameName0\\",\\"transactionDetID\\":\\"zW90\\",\\"turnover\\":\\"4.0000000\\"}"
                      />
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testStakeTurnover"
                        data-testid="StakeTurnover"
                        row="{\\"rowNo\\":1,\\"betLinkFormat\\":\\"hasBetLink0\\",\\"commission\\":\\"0.0000000\\",\\"dateTime\\":\\"2021-09-08 11:19:51.0000000\\",\\"event\\":\\"W0\\",\\"gameID\\":\\"000140\\",\\"gameName\\":\\"Djap Go0\\",\\"grossRake\\":\\"0.0000000\\",\\"memWLCommisionFormatter\\":\\"withMemWLCommission0\\",\\"netwin\\":\\"10000.0000000\\",\\"reasonFormat\\":\\"noReason0\\",\\"roundDetID\\":2,\\"roundIDFormat\\":\\"roundDetID0\\",\\"stake\\":\\"4.0000000\\",\\"tableNameFormat\\":\\"gameName0\\",\\"transactionDetID\\":\\"zW90\\",\\"turnover\\":\\"4.0000000\\"}"
                      >
                        <div>
                          4.0000000
                        </div>
                        <div>
                          4.0000000
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testGrossRake"
                        data-testid="GrossRake"
                      >
                        <div>
                          {"rowNo":1,"betLinkFormat":"hasBetLink0","commission":"0.0000000","dateTime":"2021-09-08 11:19:51.0000000","event":"W0","gameID":"000140","gameName":"Djap Go0","grossRake":"0.0000000","memWLCommisionFormatter":"withMemWLCommission0","netwin":"10000.0000000","reasonFormat":"noReason0","roundDetID":2,"roundIDFormat":"roundDetID0","stake":"4.0000000","tableNameFormat":"gameName0","transactionDetID":"zW90","turnover":"4.0000000"}
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testMembersWLCommision"
                        data-testid="MembersWLCommision"
                        row="{\\"rowNo\\":1,\\"betLinkFormat\\":\\"hasBetLink0\\",\\"commission\\":\\"0.0000000\\",\\"dateTime\\":\\"2021-09-08 11:19:51.0000000\\",\\"event\\":\\"W0\\",\\"gameID\\":\\"000140\\",\\"gameName\\":\\"Djap Go0\\",\\"grossRake\\":\\"0.0000000\\",\\"memWLCommisionFormatter\\":\\"withMemWLCommission0\\",\\"netwin\\":\\"10000.0000000\\",\\"reasonFormat\\":\\"noReason0\\",\\"roundDetID\\":2,\\"roundIDFormat\\":\\"roundDetID0\\",\\"stake\\":\\"4.0000000\\",\\"tableNameFormat\\":\\"gameName0\\",\\"transactionDetID\\":\\"zW90\\",\\"turnover\\":\\"4.0000000\\"}"
                      >
                        <div>
                          10000.0000000
                        </div>
                        <div>
                          0.0000000
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      2
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testGameDetails"
                        data-testid="GameDetails"
                        row="{\\"rowNo\\":2,\\"betLinkFormat\\":\\"hasBetLink0\\",\\"commission\\":\\"0.0000000\\",\\"dateTime\\":\\"2021-09-08 11:19:51.0000000\\",\\"event\\":\\"L0\\",\\"gameID\\":\\"000140\\",\\"gameName\\":\\"Djap Go0\\",\\"grossRake\\":\\"0.0000000\\",\\"memWLCommisionFormatter\\":\\"withMemWLCommission0\\",\\"netwin\\":\\"10000.0000000\\",\\"reasonFormat\\":\\"noReason0\\",\\"roundDetID\\":2,\\"roundIDFormat\\":\\"roundDetID0\\",\\"stake\\":\\"4.0000000\\",\\"tableNameFormat\\":\\"gameName0\\",\\"transactionDetID\\":\\"zW90\\",\\"turnover\\":\\"4.0000000\\"}"
                      >
                        <span>
                          {"rowNo":2,"betLinkFormat":"hasBetLink0","commission":"0.0000000","dateTime":"2021-09-08 11:19:51.0000000","event":"L0","gameID":"000140","gameName":"Djap Go0","grossRake":"0.0000000","memWLCommisionFormatter":"withMemWLCommission0","netwin":"10000.0000000","reasonFormat":"noReason0","roundDetID":2,"roundIDFormat":"roundDetID0","stake":"4.0000000","tableNameFormat":"gameName0","transactionDetID":"zW90","turnover":"4.0000000"}
                        </span>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testGameResult"
                        data-testid="GameResult"
                        row="{\\"rowNo\\":2,\\"betLinkFormat\\":\\"hasBetLink0\\",\\"commission\\":\\"0.0000000\\",\\"dateTime\\":\\"2021-09-08 11:19:51.0000000\\",\\"event\\":\\"L0\\",\\"gameID\\":\\"000140\\",\\"gameName\\":\\"Djap Go0\\",\\"grossRake\\":\\"0.0000000\\",\\"memWLCommisionFormatter\\":\\"withMemWLCommission0\\",\\"netwin\\":\\"10000.0000000\\",\\"reasonFormat\\":\\"noReason0\\",\\"roundDetID\\":2,\\"roundIDFormat\\":\\"roundDetID0\\",\\"stake\\":\\"4.0000000\\",\\"tableNameFormat\\":\\"gameName0\\",\\"transactionDetID\\":\\"zW90\\",\\"turnover\\":\\"4.0000000\\"}"
                      />
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testStakeTurnover"
                        data-testid="StakeTurnover"
                        row="{\\"rowNo\\":2,\\"betLinkFormat\\":\\"hasBetLink0\\",\\"commission\\":\\"0.0000000\\",\\"dateTime\\":\\"2021-09-08 11:19:51.0000000\\",\\"event\\":\\"L0\\",\\"gameID\\":\\"000140\\",\\"gameName\\":\\"Djap Go0\\",\\"grossRake\\":\\"0.0000000\\",\\"memWLCommisionFormatter\\":\\"withMemWLCommission0\\",\\"netwin\\":\\"10000.0000000\\",\\"reasonFormat\\":\\"noReason0\\",\\"roundDetID\\":2,\\"roundIDFormat\\":\\"roundDetID0\\",\\"stake\\":\\"4.0000000\\",\\"tableNameFormat\\":\\"gameName0\\",\\"transactionDetID\\":\\"zW90\\",\\"turnover\\":\\"4.0000000\\"}"
                      >
                        <div>
                          4.0000000
                        </div>
                        <div>
                          4.0000000
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testGrossRake"
                        data-testid="GrossRake"
                      >
                        <div>
                          {"rowNo":2,"betLinkFormat":"hasBetLink0","commission":"0.0000000","dateTime":"2021-09-08 11:19:51.0000000","event":"L0","gameID":"000140","gameName":"Djap Go0","grossRake":"0.0000000","memWLCommisionFormatter":"withMemWLCommission0","netwin":"10000.0000000","reasonFormat":"noReason0","roundDetID":2,"roundIDFormat":"roundDetID0","stake":"4.0000000","tableNameFormat":"gameName0","transactionDetID":"zW90","turnover":"4.0000000"}
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testMembersWLCommision"
                        data-testid="MembersWLCommision"
                        row="{\\"rowNo\\":2,\\"betLinkFormat\\":\\"hasBetLink0\\",\\"commission\\":\\"0.0000000\\",\\"dateTime\\":\\"2021-09-08 11:19:51.0000000\\",\\"event\\":\\"L0\\",\\"gameID\\":\\"000140\\",\\"gameName\\":\\"Djap Go0\\",\\"grossRake\\":\\"0.0000000\\",\\"memWLCommisionFormatter\\":\\"withMemWLCommission0\\",\\"netwin\\":\\"10000.0000000\\",\\"reasonFormat\\":\\"noReason0\\",\\"roundDetID\\":2,\\"roundIDFormat\\":\\"roundDetID0\\",\\"stake\\":\\"4.0000000\\",\\"tableNameFormat\\":\\"gameName0\\",\\"transactionDetID\\":\\"zW90\\",\\"turnover\\":\\"4.0000000\\"}"
                      >
                        <div>
                          10000.0000000
                        </div>
                        <div>
                          0.0000000
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      3
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testGameDetails"
                        data-testid="GameDetails"
                        row="{\\"rowNo\\":3,\\"betLinkFormat\\":\\"hasBetLink0\\",\\"commission\\":\\"0.0000000\\",\\"dateTime\\":\\"2021-09-08 11:19:51.0000000\\",\\"event\\":\\"L0\\",\\"gameID\\":\\"000140\\",\\"gameName\\":\\"Djap Go0\\",\\"grossRake\\":\\"0.0000000\\",\\"memWLCommisionFormatter\\":\\"withMemWLCommission0\\",\\"netwin\\":\\"10000.0000000\\",\\"reasonFormat\\":\\"noReason0\\",\\"roundDetID\\":2,\\"roundIDFormat\\":\\"roundDetID0\\",\\"stake\\":\\"4.0000000\\",\\"tableNameFormat\\":\\"gameName0\\",\\"transactionDetID\\":\\"zW90\\",\\"turnover\\":\\"4.0000000\\"}"
                      >
                        <span>
                          {"rowNo":3,"betLinkFormat":"hasBetLink0","commission":"0.0000000","dateTime":"2021-09-08 11:19:51.0000000","event":"L0","gameID":"000140","gameName":"Djap Go0","grossRake":"0.0000000","memWLCommisionFormatter":"withMemWLCommission0","netwin":"10000.0000000","reasonFormat":"noReason0","roundDetID":2,"roundIDFormat":"roundDetID0","stake":"4.0000000","tableNameFormat":"gameName0","transactionDetID":"zW90","turnover":"4.0000000"}
                        </span>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testGameResult"
                        data-testid="GameResult"
                        row="{\\"rowNo\\":3,\\"betLinkFormat\\":\\"hasBetLink0\\",\\"commission\\":\\"0.0000000\\",\\"dateTime\\":\\"2021-09-08 11:19:51.0000000\\",\\"event\\":\\"L0\\",\\"gameID\\":\\"000140\\",\\"gameName\\":\\"Djap Go0\\",\\"grossRake\\":\\"0.0000000\\",\\"memWLCommisionFormatter\\":\\"withMemWLCommission0\\",\\"netwin\\":\\"10000.0000000\\",\\"reasonFormat\\":\\"noReason0\\",\\"roundDetID\\":2,\\"roundIDFormat\\":\\"roundDetID0\\",\\"stake\\":\\"4.0000000\\",\\"tableNameFormat\\":\\"gameName0\\",\\"transactionDetID\\":\\"zW90\\",\\"turnover\\":\\"4.0000000\\"}"
                      />
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testStakeTurnover"
                        data-testid="StakeTurnover"
                        row="{\\"rowNo\\":3,\\"betLinkFormat\\":\\"hasBetLink0\\",\\"commission\\":\\"0.0000000\\",\\"dateTime\\":\\"2021-09-08 11:19:51.0000000\\",\\"event\\":\\"L0\\",\\"gameID\\":\\"000140\\",\\"gameName\\":\\"Djap Go0\\",\\"grossRake\\":\\"0.0000000\\",\\"memWLCommisionFormatter\\":\\"withMemWLCommission0\\",\\"netwin\\":\\"10000.0000000\\",\\"reasonFormat\\":\\"noReason0\\",\\"roundDetID\\":2,\\"roundIDFormat\\":\\"roundDetID0\\",\\"stake\\":\\"4.0000000\\",\\"tableNameFormat\\":\\"gameName0\\",\\"transactionDetID\\":\\"zW90\\",\\"turnover\\":\\"4.0000000\\"}"
                      >
                        <div>
                          4.0000000
                        </div>
                        <div>
                          4.0000000
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testGrossRake"
                        data-testid="GrossRake"
                      >
                        <div>
                          {"rowNo":3,"betLinkFormat":"hasBetLink0","commission":"0.0000000","dateTime":"2021-09-08 11:19:51.0000000","event":"L0","gameID":"000140","gameName":"Djap Go0","grossRake":"0.0000000","memWLCommisionFormatter":"withMemWLCommission0","netwin":"10000.0000000","reasonFormat":"noReason0","roundDetID":2,"roundIDFormat":"roundDetID0","stake":"4.0000000","tableNameFormat":"gameName0","transactionDetID":"zW90","turnover":"4.0000000"}
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testMembersWLCommision"
                        data-testid="MembersWLCommision"
                        row="{\\"rowNo\\":3,\\"betLinkFormat\\":\\"hasBetLink0\\",\\"commission\\":\\"0.0000000\\",\\"dateTime\\":\\"2021-09-08 11:19:51.0000000\\",\\"event\\":\\"L0\\",\\"gameID\\":\\"000140\\",\\"gameName\\":\\"Djap Go0\\",\\"grossRake\\":\\"0.0000000\\",\\"memWLCommisionFormatter\\":\\"withMemWLCommission0\\",\\"netwin\\":\\"10000.0000000\\",\\"reasonFormat\\":\\"noReason0\\",\\"roundDetID\\":2,\\"roundIDFormat\\":\\"roundDetID0\\",\\"stake\\":\\"4.0000000\\",\\"tableNameFormat\\":\\"gameName0\\",\\"transactionDetID\\":\\"zW90\\",\\"turnover\\":\\"4.0000000\\"}"
                      >
                        <div>
                          10000.0000000
                        </div>
                        <div>
                          0.0000000
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
